docs/quickstart.py

```diff
@@ -79,40 +177,65 @@ def cross_entropy(out, Y):
   y = Tensor(y)
   return out.mul(y).mean()
 
+# As we can see in this implementation of cross entropy loss,
+# there are certain operations that tinygrad does not support.
+# Namely, operations that are load/store like indexing a tensor with another tensor
+# or assigning a value to a tensor at a certain index.
+# Load/store ops are not supported in tinygrad because they add complexity
+# when trying to port to different backends and 90% of the models out there don't use/need them.
+
+#%%
+# == Example: setup optimizer ==
+
+# For our optimizer we will be using the traditional stochastic gradient descent optimizer with a learning rate of 3e-4.
 from tinygrad.nn.optim import SGD
 
 opt = SGD([net.l1.weight, net.l2.weight], lr=3e-4)
 
+# We can see that we are passing in the parameters of our neural network to the optimizer.
+# This is due to the fact that the optimizer needs to know which parameters to update.
+# The parameters are just listed out explicitly here for clarity.
+# There is a simpler way to do this just by using `get_parameters(net)` from `tinygrad.nn.optim`
+# which will return a list of all the parameters in the neural network like below.
+
 #from tinygrad.nn.optim import get_parameters
 #opt = SGD(get_parameters(net), lr=3e-4)
 
+#%%
+# == Example: fecthing mnist dataset ==
+
+# Now that we have our network, loss function, and optimizer defined
+# all we are missing is the data to train on!
+# There are a couple of dataset loaders in tinygrad located in [/datasets](/datasets).
+# We use the MNIST dataset loader and assign the dataset as four separate variables: `X_train`, `Y_train`, `X_test`, and `Y_test`.
+# The dataset contains our training data `X_` will be images and `Y_` will be labels
 from datasets import fetch_mnist
 
 X_train, Y_train, X_test, Y_test = fetch_mnist()
 
-def run_net(X,Y,bs):
+#%%
+# == Example: training the TinyNet ==
+
+# Now we have everything we need to start training our neural network.
+# We will be training for a target loss of 0.03 with a batch size of 64.
+from tqdm import tqdm
+
+target_loss = 0.03
+tstep, step_loss, bs = 0, 1., 64
+tbar = tqdm(total=0, initial=0, ncols=0, bar_format='Training: {desc}')
+st = time.perf_counter()
+X,Y = X_train, Y_train
+while step_loss > target_loss:
   # random sample a batch
   samp = np.random.randint(0, X.shape[0], size=(bs))
   batch = Tensor(X[samp], requires_grad=False)
   # get the corresponding labels
   labels = Y[samp]
   # forward pass
   out = net(batch)
-  return labels, out
-
-target_loss = 0.03
-tstep, losst, bs = 0, 1., 64
-
-from tqdm import tqdm
-
-tbar = tqdm(total=0, initial=0, ncols=0, bar_format='Training: {desc}')
-st = time.perf_counter()
-
-while losst > target_loss:
-  labels, out = run_net(X_train, Y_train, bs)
   # compute loss
-  #loss = cross_entropy(out, labels)
-  loss = sparse_categorical_crossentropy(out, labels)
+  loss = cross_entropy(out, labels)
+  #loss = sparse_categorical_crossentropy(out, labels)
```

Comment:
I would say just remove this line and add a comment above saying that `cross_entropy` is the same as `sparse_categorical_crossentropy` from `extra.training`

Reply:
added comment